File: Blazor server app/wwwroot/js/geoLocation.js
window.blazorGeolocation = {
    toSerializeable: function (e) {
        return {
            "coords": {
                "latitude": e.coords.latitude,
                "longitude": e.coords.longitude
            },
            "timestamp": new Date(e.timestamp)
        };
    },
    hasGeolocaitonFeature: function () {
        return navigator.geolocation ? true : false;
    },
    //meth
    getCurrentPosition: function (geolocationRef, options) {

        function onSuccess(result) {
            return geolocationRef.invokeMethodAsync('RaiseOnGetPosition',
                blazorGeolocation.toSerializeable(result));
        };
       
        function onError(er) {
            return geolocationRef
                .invokeMethodAsync('RaiseOnGetPositionError', er.code);
        };
        navigator.geolocation
            .getCurrentPosition(onSuccess, onError, options);
    },

};

File: Blazor server app/Data/GeoLocation/Geolocation.cs
using Microsoft.JSInterop;

namespace Blazor_server_app.Data.GeoLocation
{
    public class Geolocation
    {
        private readonly IJSRuntime js;
        public Geolocation(IJSRuntime js)
        {
            this.js = js;
        }
        public async ValueTask<bool> HasGeolocationFeature() =>await js.InvokeAsync<bool>("blazorGeolocation.hasGeolocaitonFeature");
        private Action<Position> OnGetPosition;
        [JSInvokable]
        public void RaiseOnGetPosition(Position p) =>OnGetPosition?.Invoke(p);

        private Action<PositionError> OnGetPositionError;
        [JSInvokable]
        public void RaiseOnGetPositionError(PositionError err) =>OnGetPositionError?.Invoke(err);
        public async ValueTask GetCurrentPosition(Action<Position> onSuccess,Action<PositionError> onError, PositionOptions options = null)
        {
                    OnGetPosition = onSuccess;
                    OnGetPositionError = onError;
                    await js.InvokeVoidAsync
                   ("blazorGeolocation.getCurrentPosition",
                    DotNetObjectReference.Create(this), options);
        }
    }
}
